### DISCUSSION TITLE
给图片写了一个放大缩小

### DISCUSSION THREAD

**Discussion by streesobrs** [BYJRK/DotNet-Discussions#63] [WPF 开发]
> 我写了一个mousewheel事件

private void Image_MouseWheel(object sender, MouseWheelEventArgs e)

        {
            int direction = Math.Sign(e.Delta);

            double scale = direction > 0 ? 1.1 : 0.9;

            Transform transform = (sender as Image).RenderTransform;

            ScaleTransform newTransform = new ScaleTransform(scale, scale);

            if (transform is TransformGroup)
            {
                (transform as TransformGroup).Children.Add(newTransform);
            }
            else if (transform != null)
            {
                TransformGroup group = new TransformGroup();
                group.Children.Add(transform);
                group.Children.Add(newTransform);
                (sender as Image).RenderTransform = group;
            }
            else
            {
                (sender as Image).RenderTransform = newTransform;
            }
        }
但是最后运行了一滚动鼠标滚轮就报错
System.NullReferenceException:“Object reference not set to an instance of an object.”
不清楚缺少什么赋值

**Comment by eijjxx** [comment]
你看下这个报错对应的是哪一行啊=.=
